go.mod

```diff
@@ -15,9 +15,9 @@ require (
 	github.com/spf13/viper v1.12.0
 	github.com/stretchr/testify v1.8.2
 	go.uber.org/goleak v1.1.12
-	k8s.io/api v0.26.10
-	k8s.io/apimachinery v0.26.10
-	k8s.io/client-go v0.26.10
+	k8s.io/api v0.27.7
+	k8s.io/apimachinery v0.27.7
+	k8s.io/client-go v0.27.7
 	k8s.io/metrics v0.25.4
 	k8s.io/utils v0.0.0-20230209194617-a36077c30491
 	sigs.k8s.io/controller-runtime v0.12.2
```

Comment:
`sigs.k8s.io/controller-runtime` has to be kept in-sync with the `k8s.io` deps. From a quick glance I can see that we'll need to use the `v0.15` version. However, there are a bunch of breaking changes that need to be revised before upgrading the controller-runtime lib. Check the details here: https://github.com/kubernetes-sigs/controller-runtime/releases